### PR TITLE
Style refactoring

### PR DESCRIPTION
**Pull request by alecarn**
This is a big one, we refactored all the library style. 
- No more magic import each file know what to do. 
- Remove scss-bundle
- Refactor the pipeline and migrate gulp script to use the native ng-packagr capabilities or node
- Allow style composition
- Remove gulp fix for @angular/cdk styles

**TO TEST THIS PR, YOU NEED TO:**
1. Check this branch
2. Delete the node_modules folder or @angular and stylus folder (unwanted fix is apply to @angular and stylus library)
3. Launch a clean install "npm ci"

**What is the current behavior?** (You can also link to an open issue here)
All the prebuilt themes was bundled and loaded in each app

**What is the new behavior?**
Compile prebuilt themes seperately and load only the theme we need.

**Does this PR introduce a breaking change?** (check one with "x")
- [x ] Yes
- [ ] No

**If this PR contains a breaking change, please describe the impact and migration path for existing applications:**
The theming configuration has change. We need to check each application, who are not using prebuilt-themes. If your application configure manually the theme, please see the file demo/src/style.scss in the Demo and then the theme config

**Other information**:

### CODE REVIEW

**Comment by aziz-access** [comment]
The Font-family not loaded also the font-size changed you can see it in the home page.

The "mat-toolbar" height changed `.mat-toolbar-row, .mat-toolbar-single-row { height: 64px;}`.

-------
In "draw/measure/import-export/spatial-filter": the color and size of buttons 'mat-button-toggle' of the component 'mat-button-toggle-group' was changed:
the background color changed from `background-color: #ffd740;` to `background-color: #e0e0e0;`
From:
image
To:
image

you need to check "mat-button-toggle-group" and "mat-button-toggle" styles.

-------
For "auth-form" component i found this
image

**Comment by alecarn** [comment]
1. Normalize the font-family and avoir flickering (DONE)
2. Mat-toolbar height is not the same (NOT DONE, This is the default style of Material. It's look like a old regression
3. Button-toggle the accent color not apply (DONE)
4. Authentication background should not over the sidenav (DONE)

File: packages/geo/src/lib/map/menu-button/menu-button.component.ts [hunk 1]
```diff
@@ -15,39 +16,37 @@ export class MenuButtonComponent {
...
   }
   private _sidenavOpenend: boolean;
 
   @Output() openSidenav = new EventEmitter<any>();
 
-  public menuButtonReverseColor = false;
+  public menuButtonReverseColor: boolean;
 
-  public menuButtonClass;
+  public menuButtonClass$: BehaviorSubject<{ [key: string]: any }> = new BehaviorSubject(undefined);
```
Comment (alecarn) [inline packages/geo/src/lib/map/menu-button/menu-button.component.ts:28]: Pourquoi un Observable?
En théorie on devrait tout avoir à notre disposition dans le constructeur pour calculer la valeur?

**Review by pelord — APPROVED** [review]

File: packages/core/ng-package.prod.json [hunk 2]
```diff
@@ -4,5 +4,14 @@
...
-  "allowedNonPeerDependencies": ["@ngx-translate"]
+  "allowedNonPeerDependencies": ["@ngx-translate"],
+  "assets": [
+    { "input": "src", "glob": "*.scss", "output": "" },
+    { "input": "src/lib", "glob": "**/*.theming.scss", "output": "lib" },
+    { "input": "src/assets", "glob": "**/*", "output": "assets" },
+    { "input": "src/style/partial", "glob": "**/*.scss", "output": "style/partial" },
+    { "input": "src/theming", "glob": "*.scss", "output": "theming" },
+    { "input": "src/locale", "glob": "*", "output": "locale" },
+    { "input": "node_modules/@mdi/angular-material", "glob": "mdi.svg", "output": "assets/icons" }
```
[the remark this answers is not in the document]
Comment (alecarn) [reply packages/core/ng-package.prod.json]: Un oublie, on ne peut pas importer des fichiers en dehors d'un package avec ng-packagr, Le script postbuild-core est là pour cette raison

**Review by pelord — APPROVED** [review]

**Pull request merged**